cmd/juju/application/deploy.go

```diff
@@ -387,80 +387,86 @@ type DeployCommand struct {
-  mediawiki-single
-  bundle/mediawiki-single
+A versioned charm URL will be expanded as expected. For example, 'mysql-56'
+becomes 'cs:xenial/mysql-56'.
 
-The current series for charms is determined first by the 'default-series' model
-setting, followed by the preferred series for the charm in the charm store.
+A bundle can be expressed similarly, but not by series:
 
-In these cases, a versioned charm URL will be expanded as expected (for
-example, mysql-33 becomes cs:precise/mysql-33).
+  juju deploy mediawiki-single
+  juju deploy bundle/mediawiki-single
+  juju deploy cs:bundle/mediawiki-single
 
-Charms may also be deployed from a user specified path. In this case, the path
-to the charm is specified along with an optional series.
+A local charm may be deployed by specifying the path to its directory:
 
-  juju deploy /path/to/charm --series trusty
+  juju deploy /path/to/charm --series xenial
 
-If '--series' is not specified, the charm's default series is used. The default
-series for a charm is the first one specified in the charm metadata. If the
-specified series is not supported by the charm, this results in an error,
-unless '--force' is used.
+A local bundle may be deployed by specifying the path to its YAML file:
 
-  juju deploy /path/to/charm --series wily --force
+  juju deploy /path/to/bundle.yaml
 
-Charms can also utilise LXD Profiles when deploying a charm. LXD Profiles can
-be used to override configurations or devices when creating the containers for
-LXD. LXD Profiles are validated against a allow/deny list, using '--force' flag
-can bypass the validation.
+A series is determined using an order of precedence (most preferred to least):
 
-Using the '--force' flag for LXD Profiles is not generally recommended when
-deploying an application; overriding profiles on the container may cause
-unexpected behavior.
+ - the '--series' command option
+ - the series stated in the charm URL
+ - for a bundle, the series stated in each charm URL (of the bundle file)
+ - for a bundle, the series given at the top level (of the bundle file)
+ - the 'default-series' model key
+ - the top-most series specified in the charm's metadata file
+   (this sets the charm's 'preferred series' in the Charm Store)
+ 
+An error is emitted if the determined series is not supported by the charm. Use
+the '--force' option to override this check:
 
-Local bundles are specified with a direct path to a bundle.yaml file.
-For example:
+  juju deploy charm --series xenial --force
 
-  juju deploy /path/to/bundle/openstack/bundle.yaml
+When charms that include LXD profiles are deployed the profiles are validated
+for security purposes by allowing only certain configurations and devices. Use
+the '--force' option to bypass this check. Doing so is not recommended as it
+can lead to unexpected behaviour.
+
+An 'application name' provides an alternate name for the application. It works
+only for charms; it is silently ignored for bundles. Such a name must consist
```

Comment:
A bundle can have an 'application name' for a charm and it is used.  The openstack bundles rename the percona-cluster charm to mysql.

Reply:
Right. I meant when used with the `deploy` command. You mean using an application name within the bundle itself right?

Reply:
Right, app names can be specified in the bundle, but I think this comment was describing the `application name` parameter for the `deploy` command:
```
$ juju deploy --help | head
Usage: juju deploy [options] <charm or bundle> [<application name>]
```

Type whatever you want for that last param when deploying a bundle; it'll be ignored (at least in my quick check).